src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs

```diff
@@ -16,16 +16,19 @@ public static string GetKeyFromRequest(AuthenticationRequestParameters requestPa
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenSilent)
             {
-                return requestParameters.Account?.HomeAccountId?.Identifier;
+                return requestParameters.Account?.HomeAccountId?.Identifier == null ? 
```

Comment:
@jmprieur - if you could review just this file, it captures all the actual changes, the rest are tests ...